core/ui/src/main/java/com/yapp/core/ui/util/DateTime.kt

```
@@ -0,0 +1,60 @@
+        }
+    } catch (e: Exception) {
+        time
+    }
+}
+
+fun isPastDate(dateString: String, pattern: String = "yyyy-MM-dd"): Boolean {
+    return try {
+        val formatter = DateTimeFormatter.ofPattern(pattern)
+        val inputDate = LocalDate.parse(dateString, formatter)
+        inputDate.isBefore(LocalDate.now())
+    } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)